internal/controller/progressive/progressive.go

```diff
@@ -392,24 +395,27 @@ func processUpgradingChild(
 
 			numaLogger.WithValues("old child", existingUpgradingChildDef.GetName(), "new child", newUpgradingChildDef.GetName()).Debug("replacing 'upgrading' child")
 			reasonFailure := common.LabelValueProgressiveFailure
+			// mark recyclable the existing upgrading child
 			err = ctlrcommon.UpdateUpgradeState(ctx, c, common.LabelValueUpgradeRecyclable, &reasonFailure, existingUpgradingChildDef)
 			if err != nil {
 				return false, 0, err
 			}
 		}
 		childStatus = rolloutObject.GetUpgradingChildStatus()
 
-		// do post-upgrade process if we haven't (check that AssessmentResult is not set just in case)
+		// After creating the new Upgradng child, do post-upgrade process (check that AssessmentResult is not set just in case) and return
 		if !childStatus.InitializationComplete && childStatus.AssessmentResult == apiv1.AssessmentResultUnknown {
 
-			needsRequeue, err := startPostUpgradeProcess(ctx, rolloutObject, existingPromotedChildDef, existingUpgradingChildDef, controller, c)
+			needsRequeue, err := startPostUpgradeProcess(ctx, rolloutObject, existingPromotedChildDef, newUpgradingChildDef, controller, c)
```

Comment:
this function needs to operate on the newly created Upgrading child. We were accidentally operating on the existing Upgrading child